applications/nrf_desktop/nrf_desktop_integration.rst

```diff
@@ -0,0 +1,953 @@
+The plugin associated with the DFU protocol realized through the :ref:`nrf_desktop_config_channel` is branded as ``nordic_hid``.
+
+Adding a new device
+===================
+
+The device specifies which protocol is used for the communication by adding the specific device information to the plugin metadata.
+For the ``nordic_hid`` plugin, this file is located at `nordic-hid.quirk`_.
+The following example shows the information passed to the plugin metadata:
+
+.. code-block:: console
+
+   [HIDRAW\VEN_1915&DEV_52DE]
+   Plugin = nordic_hid
+   GType = FuNordicHidCfgChannel
+   NordicHidBootloader = B0
+
+In this console snippet:
+
+* ``[HIDRAW\VEN_1915&DEV_52DE]`` - This line describes the device instance ID provided by the OS, which identifies the device.
+* ``Plugin = nordic_hid`` and ``GType = FuNordicHidCfgChannel`` - These lines set the plugin that the device uses.
+* ``NordicHidBootloader`` - This optional line selects the bootloader that the device is running.
+  If the device does not have the information about the underlying bootloader, the ``NordicHidBootloader`` option is used to select a proper bootloader type.
+  If there is no information about the bootloader, both in metadata and from the device, the update procedure will fail.
+  The possible values are either ``B0``, ``MCUBOOT`` or ``MCUBOOT+XIP``.
+
+.. note::
+   As the ``nordic_hid`` plugin communicates with the device using the Configuration channel, the device update is not allowed through the Serial recovery DFU.
+
+To add a new device, a pull request must be opened to the `fwupd`_ repository with a new entry to the :file:`nordic-hid.quirk` file.
+
+LVFS and update image preparation
+=================================
+
+The `LVFS (Linux Vendor Firmware Service) <LVFS_>`_ hosts firmware images that can be downloaded by Linux machines and used by the fwupd tool for the firmware update of compatible devices.
+A vendor account is needed to upload a new firmware archive to the site.
+Information on how to apply for an account is found at the `LVFS Getting an account`_ website.
+
+The nRF Desktop application DFU image is delivered as a zip package, containing a manifest and one or more binary files used for the update.
+To prepare the image file compatible with the LVFS, the CAB file needs to be prepared.
+The CAB package must contain the DFU package generated by the |NCS|, that is :file:`dfu_application.zip`, plus metadata file with information used by the LVFS.
+For more information, see the `LVFS metadata`_ site.
+
+When the CAB archive has been built, it can be uploaded to the LVFS where the CAB archive is verified and signed.
+For more information about creating CAB files, signing, and uploading the update package, see the `LVFS Uploading Firmware`_ site.
+
+Upgrading firmware using fwupd
+==============================
+
+Once the update image was uploaded onto the LVFS, the firmware update procedure can be tested on end machines.
```

Comment:
```suggestion
Once the update image is uploaded onto the LVFS, you can test the firmware update procedure on end machines.
```
End machines sounds weird. Can we find something better?

Reply:
Consider:
`Once the update image is uploaded onto the LVFS, you can test the firmware update procedure on HID hosts that support fwupd.`